src/requests/addresses/models.ts

```diff
@@ -0,0 +1,534 @@
+   */
+  "internalData"?: any,
+  /**
+   * Indicates if this is the default address for the User
+   */
+  "main": boolean,
+  /**
+   * <date-time>
+   * Datestring of the time the address was created
+   */
+  "created": string
+}
+
+export interface GetAddressInfoQueryParametersModel {
+  /**
+   * If true then resolves also wildcard addresses
+   */
+  allowWildcard: boolean;
+}
+
+export interface GetAddressInfoResponseModel {
+  /**
+   * Indicates successful response
+   */
+  "success": boolean,
+  /**
+   * ID of the Address
+   */
+  "id": string,
+  /**
+   * E-mail address string
+   */
+  "address": string,
+  /**
+   * Identity name
+   */
+  "name": string,
+  /**
+   * ID of the user if the address belongs to a User
+   */
+  "user": string,
+  /**
+   * List of forwarding targets if this is a Forwarded address
+   */
+  "targets": string[],
+  /**
+   * Account limits and usage
+   */
+  "limits": {
+    /**
+     * Forwarding quota
+     */
+    "forwards": LimitUsageTTLModel
+  },
+  /**
+   * Autoreply information
+   */
+  "autoreply": AutoreplyModel,
+  /**
+   * List of tags associated with the Address
+   */
+  "tags": string[],
+  /**
+   * Datestring of the time the address was created
+   */
+  "created": string
+}
+
+export interface ListRegisteredAddressesQueryParametersModel {
+  /**
+   * Partial match of an address
+   */
+  query?: string,
+  /**
+   * Partial match of a forward email address or URL
+   */
+  forward?: string
+  /**
+   * Comma separated list of tags. The Address must have at least one to be set
+   */
+  tags?: string;
+  /**
+   * Comma separated list of tags. The Address must have all listed tags to be set
+   */
+  requiredTags?: string;
+  /**
+   * If true, then includes metaData in the response
+   */
+  metaData?: boolean;
+  /**
+   * If true, then includes internalData in the response. Not shown for user-role tokens.
+   */
+  internalData?: boolean;
+  /**
+   * How many records to return
+   */
+  limit?: number;
+  /**
+   * Current page number. Informational only, page numbers start from 1
+   */
+  page?: number;
+  /**
+   * Cursor value for next page, retrieved from nextCursor response value
+   */
+  next?: string;
+  /**
+   * Cursor value for previous page, retrieved from previousCursor response value
+   */
+  previous?: string;
+}
+
+export interface ListRegisteredAddressesResultsModel {
+  /**
+   * ID of the Address
+   */
+  "id": string,
+  /**
+   * Identity name
+   */
+  "name": string,
+  /**
+   * E-mail address string
+   */
+  "address": string,
+  /**
+   * User ID this address belongs to if this is a User address
+   */
+  "user": string,
+  /**
+   * If true then it is a forwarded address
+   */
+  "forwarded": boolean,
+  /**
+   * If true then the forwarded address is disabled
+   */
+  "forwardedDisabled": boolean,
+  /**
+   * List of forwarding targets
+   */
+  "target"?: string[]
+}
+
+export interface ListRegisteredAddressesResponseModel {
```

Comment:
You can extend with `CursorResponseModel` so you only have to specify the results